libslide/src/bignum/compare.rs

```diff
@@ -3,53 +3,104 @@ use std::cmp::Ordering;
 
 impl Ord for Bignum {
     fn cmp(&self, other: &Bignum) -> Ordering {
+        let (lhs, rhs) = (strip_zeros(self), strip_zeros(other));
+        // lhs and rhs have no leading or trailing (in decimal portion) zeros
```

Comment:
There's no need to allocate a new bignum; just keep a small int pointing to where leading/trailing zeros (int/decimal) stand.

Reply:
Do you mean not modifying either of the Vecs at all and just altering ranges for int-by-int checks?